### PR TITLE
Add develop doc on adding new metrics

### PR DESCRIPTION
**Pull request by CatherineF-dev**
**What this PR does / why we need it**:

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)* No

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by CatherineF-dev** [comment]
cc @mrueg

File: docs/developer/guide.md [hunk 1]
```diff
@@ -20,3 +21,13 @@ The following steps are needed to introduce a new resource and its respective re
...
 - Add a sample Kubernetes manifest to be used by tests in the [tests/manifests/](https://github.com/kubernetes/kube-state-metrics/tree/main/tests/manifests) directory.
 - Lastly, and most importantly, actually implement your new resource(s) and its test binary in [internal/store](https://github.com/kubernetes/kube-state-metrics/tree/main/internal/store). Follow the formatting and structure of other resources.
+
+### Add New Metrics
+
+- Make metrics experimental first when introducing them, [Example PR](https://github.com/kubernetes/kube-state-metrics/pull/1910).
+
+| Metric stability level |                    |
+|------------------------|--------------------|
+| EXPERIMENTAL           | basemetrics.ALPHA  |
```
Comment (mrueg) [inline docs/developer/guide.md:31]: Should this stay experimental or should we rename to alpha?

**Comment by mrueg** [comment]
Thanks @CatherineF-dev for that PR! Good idea to havd some documentation around it. One small question since I believe we have used alpha and experimental in the metrics doc somewhere

**Comment by CatherineF-dev** [comment]
Yes, will rename a few `Alpha` into `Experimental` first. 

And create another issue to discuss whether converting `Alpha` to `Experimental`.

File: docs/developer/guide.md (hunk 1 above)
Comment (CatherineF-dev) [reply docs/developer/guide.md:31]: I think we can create another issue to discuss whether converting Alpha to Experimental.

From my point of view, Experimental is easier to understand while Alpha aligns with k8s stability level.

**Comment by CatherineF-dev** [comment]
Have converted one `alpha` into `experimental` in metrics doc.

**Comment by mrueg** [comment]
Thanks

/approve

File: docs/developer/guide.md [hunk 2]
```diff
@@ -20,3 +21,13 @@ The following steps are needed to introduce a new resource and its respective re
 - Reference the new resource in [pkg/options/resource.go](https://github.com/kubernetes/kube-state-metrics/blob/main/pkg/options/resource.go).
 - Add a sample Kubernetes manifest to be used by tests in the [tests/manifests/](https://github.com/kubernetes/kube-state-metrics/tree/main/tests/manifests) directory.
 - Lastly, and most importantly, actually implement your new resource(s) and its test binary in [internal/store](https://github.com/kubernetes/kube-state-metrics/tree/main/internal/store). Follow the formatting and structure of other resources.
+
+### Add New Metrics
+
+- Make metrics experimental first when introducing them, [Example PR](https://github.com/kubernetes/kube-state-metrics/pull/1910).
```
Comment (rexagod) [inline docs/developer/guide.md:27]:
```suggestion
- Make metrics experimental first when introducing them, refer [#1910](https://github.com/kubernetes/kube-state-metrics/pull/1910) for more information.
```

File: docs/developer/guide.md (hunk 1 above)
Comment (rexagod) [reply docs/developer/guide.md:31]: +1 for `experimental`, since `alpha` can hint toward more stabilty levels (`beta`, etc.) which can be misleading.

**Review by rexagod — CHANGES_REQUESTED** [review]

**Pull request merged**